spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -77,7 +78,7 @@
 	 * The AppEngine Configuration for the {@link AppEngineCloudSqlJdbcInfoProvider}.
 	 */
 	@ConditionalOnClass(com.google.cloud.sql.mysql.SocketFactory.class)
-	@Conditional(AppEngineCondition.class)
+	@ConditionalOnGcpEnvironment({GcpEnvironment.APP_ENGINE_FLEXIBLE, GcpEnvironment.APP_ENGINE_STANDARD})
```

Comment:
The original condition would only match for App Engine Standard, not Flex.
Also, I believe that with the latest version of the Cloud SQL Socket Factory, you don't even need special configuration for App Engine Standard, but I haven't validated that.

Reply:
thanks @meltsufin still trying to resolve the issue I have with the usage of `@ConditionalOnGcpEnvironment`.

> Also, I believe that with the latest version of the Cloud SQL Socket Factory, you don't even need special configuration for App Engine Standard, but I haven't validated that.

Are there any way I can validate this?

Thanks in advance.

Reply:
@kurtisvg Can you confirm that the JDBC connection string format is the same now for both App Engine Standard and Flex/GKE/etc.

Reply:
Correct - if you are using the SocketFactory, the JDBC url should be the same regardless of environment.